src/input/input_json.rs

```diff
@@ -111,70 +123,39 @@ impl<'a> Input<'a> for JsonValue {
         self.validate_bytes(false)
     }
 
-    fn strict_bool(&self) -> ValResult<bool> {
-        match self {
-            JsonValue::Bool(b) => Ok(*b),
-            _ => Err(ValError::new(ErrorTypeDefaults::BoolType, self)),
-        }
-    }
-    fn lax_bool(&self) -> ValResult<bool> {
+    fn validate_bool(&self, strict: bool) -> ValResult<'_, ValidationMatch<bool>> {
```

Comment:
i wonder if we could make `strict` here and `strict` and `coerce_numbers_to_str` generic parameters rather than function parameters in future?

Reply:
Interesting thought, I think possibly yes, the hardest bit would be dispatching to this function based on runtime strictness, we could probably work something out in a future PR.